code-examples/find-duplicate-number/src/FindDuplicate.java

```diff
@@ -0,0 +1,36 @@
+import java.util.HashSet;
+import java.util.Set;
+
+class FindDuplicate {
+    private int result;
+
+    // How do you find the duplicate number on a given integer array?
+    int findDuplicateSingle(int[] arr) {
```

Comment:
I wasn't sure if I should include this method (that is run in the first test case), because this method returns an integer that has been duplicated - and can only work **iff** there's one duplicate (as per the question). The method below can work if there are multiple duplicates.

Reply:
The second method `findDuplicateMultiple()` and related test cases supersede this method.